app/scripts/controllers/metamask-notifications/metamask-notifications.ts

```diff
@@ -552,17 +729,90 @@ export class MetamaskNotificationsController extends BaseController<
       await this.#storage.setNotificationStorage(JSON.stringify(userStorage));
 
       // Update the state of the controller
+      this.setMetamaskNotificationsFeatureSeen();
       this.setFeatureAnnouncementsEnabled(true);
-      this.setMetamaskNotificationsEnabled(true);
       this.setSnapNotificationsEnabled(true);
+      this.setMetamaskNotificationsEnabled(true);
+      this.#setIsUpdatingMetamaskNotifications(false);
 
       return userStorage;
     } catch (err) {
+      this.#setIsUpdatingMetamaskNotifications(false);
       log.error('Failed to create On Chain triggers', err);
       throw new Error('Failed to create On Chain triggers');
     }
   }
 
+  /**
+   * Enables all MetaMask notifications for the user.
+   * This method performs several key operations:
+   * 1. Validates the storage key and bearer token necessary for authentication.
+   * 2. Retrieves all linked accounts.
+   * 3. Ensures profile syncing is enabled if the storage key is missing.
+   * 4. Enables on-chain triggers for each account.
+   * 5. Sets the global notification settings for MetaMask, feature announcements, and Snap notifications to true.
+   *
+   * @throws {Error} If there is an error during the process of enabling notifications.
+   */
+  public async enableMetamaskNotifications() {
+    try {
+      this.#setIsUpdatingMetamaskNotifications(true);
+
+      await this.#performEnableProfileSyncing();
+
+      const { accounts } = await this.#accounts.listAccounts();
+
+      // For each account enable the triggers
+      await this.updateOnChainTriggersByAccount(accounts);
+
+      // Set the states
+      this.setMetamaskNotificationsEnabled(true);
+      this.setFeatureAnnouncementsEnabled(true);
+      this.setSnapNotificationsEnabled(true);
+      this.#setIsUpdatingMetamaskNotifications(false);
+    } catch (e) {
+      this.#setIsUpdatingMetamaskNotifications(false);
+      log.error('Unable to enable notifications', e);
+      throw new Error('Unable to enable notifications');
+    }
+  }
+
+  /**
+   * Disables all MetaMask notifications for the user.
+   * This method ensures that the user is authenticated, retrieves all linked accounts,
+   * and disables on-chain triggers for each account. It also sets the global notification
+   * settings for MetaMask, feature announcements, and Snap notifications to false.
+   *
+   * @throws {Error} If the user is not authenticated or if there is an error during the process.
+   */
+  public async disableMetamaskNotifications() {
+    try {
+      this.#setIsUpdatingMetamaskNotifications(true);
+
+      this.#assertAuthEnabled();
+
+      const { accounts } = await this.#accounts.listAccounts();
+
+      // For each account disable the triggers
+      await this.deleteOnChainTriggersByAccount(accounts);
```

Comment:
As discussed this is incorrect. We do not want to delete triggers if users turn off/disable notifications.

E.g. If they turn off notifications on 1 device, all other devices will stop receiving notifications.

Reply:
Future bug fix ticket.

Reply:
Ticket: https://consensyssoftware.atlassian.net/browse/NOTIFY-633